crew

```diff
@@ -725,6 +725,7 @@ def expand_dependencies
         @dependencies << dep
         search dep, true
         push_dependencies
+        @dependencies = @dependencies - [dep] + [dep]
```

Comment:
I'm not following.  Is the intention to rearrange the dep order?

Reply:
Could this be done in `push_dependencies` instead?

Reply:
This will move the dependencies to last after a loop of sub-dependency checking (`search dep, true` and `push_dependencies`)

Reply:
(I also tried to move `@dependencies << dep` to after `push_dependencies` but it causes an infinite loop)

Reply:
Can we put this logic in `push_dependencies`?